param/parameterized.py

```diff
@@ -609,6 +628,17 @@ class hierarchy (see ParameterizedMetaclass).
         self.per_instance = per_instance
 
 
+    @property
+    def label(self):
+        if self.name and self._label is None:
```

Comment:
If someone were to set name to ``''`` somehow this would still return ``self._label``, i.e. None. I don't think that matters much though since no one should ever be able to set name to ``''``.

Reply:
Right, I think it is impossible to have an empty attribute name...